### PR TITLE
Scope update

### PR DESCRIPTION
**Pull request by oleersoy**
Still trying to understand the different between long and short.  Does short mean buy it at the current price only, whereas long means the trader has more flexibility to buy at higher price and should expend the entire budget to get as much as possible?  For example if I have 1 BTC and only 10,000 XLM are available at the current price of 0.000032 BTC, but there's more available at 0.000033 BTC then the long strategy execute two orders one at 0.000032 BTC and one at 0.000033 BTC in order to spend the entire budget?

### CODE REVIEW

File: docs/introduction/scope.md
Comment (askmike) [inline docs/introduction/scope.md:7]: Besides OHLC Gekko also exposes VWP and amount of trades metrics, can we keep that in this document?

File: docs/introduction/scope.md
Comment (askmike) [inline docs/introduction/scope.md:11]: it's important to understand that the strategy does not execute any orders, it merely suggests to Gekko what to do (in the form of a position to take). Gekko has different plugins that will actually execute orders based on this advice.

Right now Gekko will always try to buy on long signals an try to sell on short signals, but the main point here is that strategies indicate "positions" to be in in. And selling and buying is right now the only way of moving from one position into another. At some point we will add margin trading which makes this less simple.

File: docs/introduction/scope.md
Comment (askmike) [inline docs/introduction/scope.md]: Why did you remove the "all-in" part? This is one of the questions asked most.

**Review by askmike — COMMENTED** [review]
See comments.

**Comment by askmike** [comment]
Thanks a lot! I really like everything, except for the short/long stuff. To answer your question:

> Still trying to understand the different between long and short. Does short mean buy it at the current price only, whereas long means the trader has more flexibility to buy at higher price and should expend the entire budget to get as much as possible?

If you are trading BTC/USD:

- short means getting into a position where you do not have any BTC (sell BTC into USD)
- long means getting into a position where you have a lot of BTC (buy BTC from USD)

When your strat signals either of the above Gekko will try to buy or sell as much as possible.

> For example if I have 1 BTC and only 10,000 XLM are available at the current price of 0.000032 BTC, but there's more available at 0.000033 BTC then the long strategy execute two orders one at 0.000032 BTC and one at 0.000033 BTC in order to spend the entire budget?

This is a different question: if your strat signals "LONG" Gekko might try to buy XLM for your BTC. Gekko won't actually buy any XLM for the prices you mentioned, but will place a "bid" order at a lower price and will wait until someone sells into your bid order. This way you get a better price (but will have to wait a bit longer).

File: docs/introduction/scope.md
Comment (oleersoy) [reply docs/introduction/scope.md:7]: Absolutely.  I tried to google VWP and did not find any documents describing what that is, so I thought it might be confusing for others as well, which is why I took it out.  Ideally there would be a link to an API document that shows how to go about utilizing the VWP and trade volume metrics.  That would also make it clear what they are since the description would be detailed in the API JS Doc / TypeDoc or perhaps a something like [Compodoc](https://medium.com/vincent-ogloblinsky/compodoc-documentation-tool-for-angular-2-applications-44ec650e01a8) would work.

File: docs/introduction/scope.md
Comment (oleersoy) [reply docs/introduction/scope.md]: My bad.  I should have started some micro issues on topics I'm confused about before updating things that I'm not 100% certain of.  I'll try to clear up my misunderstandings and then hopefully I can write better docs.  Thanks for all your patient help with this.  The tool is really awesome.

**Comment by askmike** [comment]
You are right about these docs, I am reworking the events that flow through
gekko (with proper documentation this time). A candle is one of these
events (the only one strategies can see - plugins can see everything).

If you have more questions feel free to ask them. I am already happy that
you are looking critically at the documentation!

File: docs/introduction/scope.md
Comment (oleersoy) [reply docs/introduction/scope.md]: I'm still a little confused WRT `LONG`, `SHORT`, and how orders are triggered in general.  I added an issue here to explore it further: https://github.com/askmike/gekko/issues/1686

**Comment by askmike** [comment]
@oleersoy I cannot merge this mainly because of this comment: https://github.com/askmike/gekko/pull/1683#discussion_r161392697

Strategies really do not execute orders, and I cannot accept changes to the documentation indicating that they do. Even though I really like your other changes. If you can revert that paragraph I can at least accept your other changes.

**Comment by oleersoy** [comment]
Would it be correct to say "Strategies can either trigger a `LONG` (Sell all shares) or `SHORT` (Buy up to the current budget allowance)"?

**Comment by askmike** [comment]
@oleersoy yep that's fine :)

File: docs/introduction/scope.md
Comment (askmike) [inline docs/introduction/scope.md:21]: This is definitely not strategy execution, this is the execution strategy: they are 2 different concepts and execution in this one refers to order execution.

File: docs/introduction/scope.md
Comment (askmike) [reply docs/introduction/scope.md:11]: see ^

**Review by askmike — CHANGES_REQUESTED** [review]
Last two things are quite important!

So that's:

- this: https://github.com/askmike/gekko/pull/1683#discussion_r168984524
- and this: https://github.com/askmike/gekko/pull/1683#discussion_r161392697

**Comment by askmike** [comment]
Closing this due to inactivity. I also think there will probably be some merge conflicts to figure out before we look at merging this.

Feel free to comment / push commits and we'll reopen :)

**Pull request closed without merging**
